src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextAfterAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Health.Dicom.Core.Features.Context;
+
+namespace Microsoft.Health.Dicom.Api.Features.Context
+{
+    /// <summary>
+    /// Middleware that runs after authentication middleware so that it can retrieved authenticated user claims.
+    /// </summary>
+    public class DicomRequestContextAfterAuthenticationMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public DicomRequestContextAfterAuthenticationMiddleware(RequestDelegate next)
+        {
+            EnsureArg.IsNotNull(next, nameof(next));
+
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicomRequestContextAccessor)
+        {
+            // Now the authentication is completed successfully, sets the user.
+            if (context.User != null)
```

Comment:
Should we check security enabled and fail if user is null here?

Reply:
I don't believe that would get us anything useful. At this point we're just seeing if we can decorate the `DicomRequestContext` with a user after the authentication has run. We still want the code to flow though because we have non-authenticated endpoints such as `/health/check`